feature/coupled-crow:   update model, use netcdf atm output, updates for wave post

JessicaMeixner-NOAA opened the pull request:
This PR has various updates in preparation for prototype 6 runs including: 
* Updates to use the most recent version of ufs-weather-model which required changes from "v16beta"->"v16" in the CCPP suite files and a temporary build fix for orion. 
* Does not link un-needed files during wave prep for fully coupled (Fixes Issue #236) 
* Adds separate job resources for wave post jobs (Issue #209) 
* Uses netcdf instead of nemsio output for the atm model (Issue #250) 
* Updates a few namelist variables for p6 configuration including adding dz_min, turning on nst_anl and updating FNMSKH

A test on hera has been made, see COM: 
/scratch2/NCEPDEV/climate/Jessica.Meixner/p6workflow/test02/COMROOT/test02/gfs.20130401/00 the atm grib files should be double checked by someone else. 

This PR should close the issues mentioned: #209, #236, #250

JessicaMeixner-NOAA committed:
update to latest ufs-weather-model

JessicaMeixner-NOAA committed:
add if statements in wave prep for linking to external data

This addresses Issue #236

JessicaMeixner-NOAA committed:
add seperate resources for wave post jobs
Issue #209

JessicaMeixner-NOAA committed:
update FNMSKH

JessicaMeixner-NOAA committed:
turn on nst_anl

JessicaMeixner-NOAA committed:
update to latest ufs-weather which removes beta from v16beta for ccpp

JessicaMeixner-NOAA committed:
adding dz_min variable for input.nml

this is needed after revision 6a4c307 which set dz_min=2 as default

JessicaMeixner-NOAA committed:
add cleanup line to ww3prepost build

JessicaMeixner-NOAA committed:
add temporary orion workaround for ufs build

JessicaMeixner-NOAA committed:
updates to use netcdf output instead of nemsio for atm

JessicaMeixner-NOAA committed:
updates for using netcdf instead of nemsio

Issue #250

JessicaMeixner-NOAA committed:
more ccpp updates for v16beta to v16

JessicaMeixner-NOAA committed:
updating dependency for moving from nemsio to netcdf

Issue #250

JessicaMeixner-NOAA commented:
@LydiaStefanova-NOAA could you check the atm grib files here: /scratch2/NCEPDEV/climate/Jessica.Meixner/p6workflow/test02/COMROOT/test02/gfs.20130401/00

LydiaStefanova-NOAA commented:
@JessicaMeixner-NOAA  The sfluxgrbf files look good. The rest (pgrb2.1p00, pgrb2b.1p00, flux.1p00, pgrb2.0p25 and pgrb2b.0p25) are all empty.

JessicaMeixner-NOAA commented:
Thanks @LydiaStefanova-NOAA  I converted this PR to a draft so we can find and fix this issue before merging

JessicaMeixner-NOAA committed:
update to latest ufs-weather-model and un-do temp fix for build

JessicaMeixner-NOAA committed:
remove reference to CCPP

JessicaMeixner-NOAA committed:
added a variable to control creation of ESMFProfile.Summary creation

JessicaMeixner-NOAA committed:
added config variables for seaice and lake ice minimum values

JessicaMeixner-NOAA committed:
changing OUTPUT_FILE to be just netcdf and correcting usage in model_configure

JessicaMeixner-NOAA committed:
change location of WRITE post variable so that jjob correctly
does inline post

JessicaMeixner-NOAA committed:
update coupled nceppost with what is in the ncep post lib

this means the extra creation of 1deg flux files are not made
but using netcdf and inline post now work

JessicaMeixner-NOAA committed:
change name of file exgfs_nceppost_cpl.sh.ecf -> exgfs_nceppost_cpl.sh

JessicaMeixner-NOAA committed:
update to have flux.1p00 created again

JessicaMeixner-NOAA committed:
revert savedata=yes

JessicaMeixner-NOAA commented:
@LydiaStefanova-NOAA please check: /scratch2/NCEPDEV/climate/Jessica.Meixner/p6netcdf/testnetcdf01/COMROOT/netcdftest02/gfs.20130401/00/atmos I believe all issues with atm post have been resolved.

LydiaStefanova-NOAA commented:
@JessicaMeixner-NOAA All atm grib2 files are looking good!

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA this is ready for re-review (a fair amount has changed since you gave it the thumbs up) and is ready to be merged when you think it's acceptable.

WalterKolczynski-NOAA commented on workflow/defaults/settings.yaml:
```diff
@@ -29,6 +29,7 @@ default_settings: &default_settings
   nco_mode: NO
   use_nco_ecflow_headers: NO
   print_esmf: .false.
+  esmf_profile: .flase.
```
typo

JessicaMeixner-NOAA replied:
This typo is fixed. I'm not sure where the indentation issues are, but if you point them out I will at least fix the ones in this PR.

WalterKolczynski-NOAA reviewed (changes requested):
One typo noted.

Also, be more careful with indentation in the future. I know the current codebase already is terrible with inconsistent indentation, I'd like to not make the problem worse after this (bonus points if you fix existing indentation issues in future PRs).

JessicaMeixner-NOAA committed:
fix typo

WalterKolczynski-NOAA commented on scripts/exgfs_nceppost_cpl.sh:
```diff
@@ -155,21 +164,70 @@ then
-       $DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERANL}
+       $DBNROOT/bin/dbn_alert MODEL GFS_MSC_sfcanl $job $COMOUT/${PREFIX}sfcanl${SUFFIX}
+       $DBNROOT/bin/dbn_alert MODEL GFS_SA $job $COMOUT/${PREFIX}atmanl${SUFFIX}
+#alert removed in v15.0       $DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERANL}
```
Indentation (though it can probably just be deleted)

WalterKolczynski-NOAA commented on scripts/exgfs_nceppost_cpl.sh:
```diff
@@ -302,45 +359,57 @@ do
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2B_0P25 $job $COMOUT/${PREFIX}pgrb2b.0p25.f${fhr}
-            $DBNROOT/bin/dbn_alert MODEL GFS_PGB2B_0P25_WIDX $job $COMOUT/${PREFIX}pgrb2b.0p25.f${fhr}.idx
-          fi
+            if [ $INLINE_POST = ".false." ]; then 
```
Indentation not consistent

WalterKolczynski-NOAA commented on scripts/exgfs_nceppost_cpl.sh:
```diff
@@ -302,45 +359,57 @@ do
+	if test "$SENDDBN" = 'YES'
+	then
+	    if [ $GRIBVERSION = 'grib2' ] ; then
+#alert removed in v15.0		$DBNROOT/bin/dbn_alert MODEL GFS_MASTER $job $COMOUT/${MASTERFL}
```
Indentation (remove?)

WalterKolczynski-NOAA commented on scripts/exgfs_nceppost_cpl.sh:
```diff
@@ -436,65 +499,56 @@ do
+          #   hourly if fhr<=24
+          #   every 3 forecast hour if 24<fhr<=48
+          #   every 6 forecast hour if 48<fhr<=120
+	  if [  $fhr -le 24  ] ; then
```
Indentation not consistent with previous

JessicaMeixner-NOAA replied:
This is copied from the NCEP post library -- it should just be linked in but we have to add one extra loop for the flux.1p00, therefore if the indentation should be fixed, it should be fixed in the NCEP Post library and copied in here.  This file shouldn't even exist in this repo, but it's the only work around we have for now.

WalterKolczynski-NOAA commented on scripts/exgfs_nceppost_cpl.sh:
```diff
@@ -436,65 +499,56 @@ do
 
+          if [ -e $PGBOUT ]
+          then
           if test $SENDCOM = "YES"
```
Needs to be indented from outer control block

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
+	  fhr=$FHMIN
+	  while [ $fhr -le $FHMAX ]; do
+	    FH3=$(printf %03i $fhr)
+            FH2=$(printf %02i $fhr)
```
Indentation doesn't match surrounding

JessicaMeixner-NOAA replied:
These were all issues of a mix of tabs and spaces being used, they line up when looking at the routines in vi but not when looking at it in GitHub.  All tabs were replaced by spaces in this routine.  All ush/forecast_postdet.sh indentation issues should be fixed.

WalterKolczynski-NOAA replied:
At some point, all spaces will need to be replaced with tabs in all g-w bash scripts, but that is not something you need to worry about. This is fine for now.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
+	    atmi=atmf${FH3}.$affix
+	    sfci=sfcf${FH3}.$affix
+	    logi=logf${FH3}
+            pgbi=GFSPRS.GrbF${FH2}
```
Indentation doesn't match surrounding

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
+	    atmo=$memdir/${CDUMP}.t${cyc}z.atmf${FH3}.$affix
+	    sfco=$memdir/${CDUMP}.t${cyc}z.sfcf${FH3}.$affix
+	    logo=$memdir/${CDUMP}.t${cyc}z.logf${FH3}.txt
+            pgbo=$memdir/${CDUMP}.t${cyc}z.master.grb2f${FH3}
```
Indentation doesn't match surrounding

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
+	    eval $NLN $atmo $atmi
+	    eval $NLN $sfco $sfci
+	    eval $NLN $logo $logi
+            if [ $WRITE_DOPOST = ".true." ]; then
```
Indentation doesn't match surrounding

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -435,47 +435,48 @@ FV3_GFS_postdet(){
+	  for n in $(seq 1 $ntiles); do
+	    eval $NLN nggps2d.tile${n}.nc       $memdir/nggps2d.tile${n}.nc
+	    eval $NLN nggps3d.tile${n}.nc       $memdir/nggps3d.tile${n}.nc
+            eval $NLN grid_spec.tile${n}.nc     $memdir/grid_spec.tile${n}.nc 
```
Indentation doesn't match surrounding

WalterKolczynski-NOAA commented on ush/inter_flux.sh:
```diff
@@ -45,9 +45,16 @@ else
+
+  if [ $INLINE_POST = ".false." ]; then
+    $WGRIB2 $PGBOUT $option1 $option21 $option22 $option23 $option24 \
+                          $option25 $option26 $option27 $option28 \
```
Indentation for line continuations should be consistent with each other (L51 & 52)

JessicaMeixner-NOAA replied:
should be fixed now

WalterKolczynski-NOAA reviewed (changes requested):
Indentation issues noted.

JessicaMeixner-NOAA committed:
fix spacing issues

tabs were converted to spaces in ush/forecast_postdet.sh

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA with the exception of scripts/exgfs_nceppost_cpl.sh which should be linked to ../sorc/gfs_post.fd/scripts/exgdas_atmos_nceppost.sh and therefore carries any indentation issues over from that file, the indentation issues should now be fixed.
